packages/grafana-data/src/panel/PanelPlugin.ts

```diff
@@ -259,6 +261,33 @@ export class PanelPlugin<
     return this;
   }
 
+  /**
+   * Tells Grafana if the plugin should subscribe to DashboardQueryRunner results.
+   *
+   * @example
+   * ```typescript
+   *
+   * import { ShapePanel } from './ShapePanel';
+   *
+   * interface ShapePanelOptions {}
+   *
+   * export const plugin = new PanelPlugin<ShapePanelOptions>(ShapePanel)
+   *     .useFieldConfig({})
+   *     ...
+   *     ...
+   *     .setDashboardQuerySupport({
+   *       annotations: true,
+   *       alertStates: true,
+   *     });
+   * ```
+   *
+   * @public
+   **/
+  setDashboardQuerySupport(supportOptions: Partial<DashboardQuerySupport>) {
+    this.dashboardQuerySupport = { ...this.dashboardQuerySupport, ...supportOptions };
+    return this;
+  }
+
```

Comment:
on second thought panel DashboardQuerySupport  option does not make sense (The panel should not care where the data is coming from). When we add a more general dashbord query feature it will just be a matter of subscribing to dashboard level queries, and the panel should not care if the data comes from a panel query or dashboard query. 

So I think this should probably just be setAnnotationSupport(...)   (so basically just a name change).  

thoughts @hugohaggmark ?

Reply:
@torkelo I think that makes sense.

Reply:
yea, just to clarify my thinking in that setAnnotationSupport is about informing Grafana about support for visualization annotations, whereas naming this setDashboardQuerySupport links the option to where the data is coming from (which I think panels should not have to think about).

Reply:
Makes sense to me.
I just wonder if specifying "annotation support" obscures the fact that it deals with alert states as well?

Reply:
Ah, also seems like "AnnotationSupport" is used in a few different places with a different context, so it might be confusing to use the same term here.